warehouse/localdisk.go

```diff
@@ -0,0 +1,125 @@
+package warehouse
+
+import (
+	"fmt"
+	"github.com/fullstorydev/hauser/config"
+	"github.com/nishanths/fullstory"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+type LocalDisk struct {
+	conf *config.Config
+}
+
+const (
+	timestampFile string = ".sync.hs"
+)
+
+var _ Warehouse = (*LocalDisk)(nil)
+
+func NewLocalDisk(c *config.Config) *LocalDisk {
+	if c.Local.UseStartTime {
+		filename := filepath.Join(c.Local.SaveDir, timestampFile)
+		if _, err := os.Stat(filename); !os.IsNotExist(err) {
+			os.Remove(filename)
+		}
+	}
+
+	return &LocalDisk{
+		conf: c,
+	}
+}
+
+// Currently returns the names of all the fields exported by BundleFields
+func (w *LocalDisk) GetExportTableColumns() []string {
+	allFields := BundleFields()
+	fieldNames := make([]string, 0, len(allFields))
+	for name := range allFields {
+		fieldNames = append(fieldNames, name)
+	}
+	return fieldNames
+}
+
+// Reads the last sync date from file
+func (w *LocalDisk) LastSyncPoint() (time.Time, error) {
+	t := beginningOfTime
+	if w.conf.Local.UseStartTime {
+		t = w.conf.Local.StartTime
```

Comment:
It may be good to add a validation to `NewLocalDisk` to be sure that there is a `StartTime` specified when `UseStartTime` is true? I'm not sure how missing/empty strings would get parsed from the .toml file; I suspect that you would just have a zero value for the time.Time struct.

Reply:
I think it depends on the particular toml parser implementation. I looked at the documentation for the one we're using, but didn't see anything specific to datetime. It's using reflection, so I'm guessing a missing datetime will be set to nil? I added a check to `NewLocalDisk` that will fail if `UseStartTime` is true and `StartTime` is nil.

Reply:
Actually, not nil, but the "zero" value of time

Reply:
I don't think it's possible for `StartTime` to be nil, as it's a `time.Time` struct and not a pointer to a struct (`*time.Time`). Welcome to golang! Read this article on [zero values](https://programming.guide/go/default-zero-value.html), hopefully it helps. I suspect that if the any value is missing from the toml, after parsing it will contain the zero value appropriate to its type. 

For the `time.Time` struct, its `wall` and `ext` values will be set to zero, which results in a zero value of `January 1, year 1, 00:00:00.000000000 UTC` (from the docs). Luckily it provides a handle `IsZero()` function, so you can check for a zero value, which likely means it hasn't been initialized. 

You should this out, as all of the above is just educated guessing.

Reply:
Sorry, I didn't reload the page in time to see your reply.

Reply:
Just tested it with StartTime not initialized, and it fails the IsZero check, as expected